test/test_backend/test_workers.py

```diff
@@ -60,6 +60,19 @@ def test_goto_assignments():
     assert len(results) == 0
 
 
+def test_extract_def():
+    code = """
+    import pyqode.python.widgets
+    import PyQt5.QtWidgets as QtWidgets
+    app = QtWidgets.QApplication([])
+    editor = pyqode.python.widgets.PyCyodeEdit()
+    editor.file.open(__file__)
+    editor.show()
+    app.exec()
+    """
+    results = workers._extract_def(workers.defined_names({'code': code, 'path': __file__}),"")
```

Comment:
This fails here because workers.defined_names returns a list of definitions and  workers._extract_def exects a single definition.